dashboard/test/controllers/code_reviews_controller_test.rb

```diff
@@ -0,0 +1,113 @@
+require 'test_helper'
+
+class CodeReviewsControllerTest < ActionController::TestCase
+  self.use_transactional_test_case = true
+
+  setup_all do
+    @project_owner = create :student
+    @storage_id = create_storage_id_for_user(@project_owner.id)
+    @channel_id = create :project, storage_id: @storage_id
+    _,  @project_id = storage_decrypt_channel_id(@channel_id)
+  end
+
+  setup do
+    sign_in @project_owner
+  end
+
+  test 'index returns both open and closed code reviews' do
+    script_id = 12
+    level_id = 5
+
+    closed_at = DateTime.now
+    create :code_review, user_id: @project_owner.id, project_id: @project_id,
+      script_id: script_id, level_id: level_id, closed_at: closed_at
+    create :code_review, user_id: @project_owner.id, project_id: @project_id,
+      script_id: script_id, level_id: level_id, closed_at: closed_at + 1.second
+    create :code_review, user_id: @project_owner.id, project_id: @project_id,
+      script_id: script_id, level_id: level_id, closed_at: nil
+
+    get :index, params: {
+      channelId: @channel_id,
+      scriptId: script_id,
+      levelId: level_id
+    }
+    assert_response :success
+
+    response_json = JSON.parse(response.body)
+    assert_equal 3, response_json.length
+  end
+
+  test 'create code review' do
```

Comment:
Could also test that trying to create a code review on a project that you don't own fails.

Reply:
Oh I just saw this is tested in the ability test